Tests/LibC/TestMemalign.cpp

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2022, Peter Elliott <pelliott@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibTest/TestCase.h>
+
+#include <LibC/mallocdefs.h>
+#include <stdlib.h>
+
+static constexpr size_t runs = 5000;
+static constexpr size_t ptrs_per_run = 20;
+
+static size_t random_alignment()
+{
+    return 1 << (rand() % 16);
+}
+
+static size_t random_size()
+{
+    int r = rand() % num_size_classes;
+    if (r == num_size_classes - 1) {
+        return rand() % (1 << 17);
+    }
+
+    return size_classes[r] + (rand() % (size_classes[r + 1] - size_classes[r]));
+}
```

Comment:
Personally, I'd like tests to have as few random variables to it as possible. This test is intentionally testing a number of random calls to see if they succeed, but if some values fail it would take some time before CI starts randomly failing.

I'd suggest selecting a couple of (edge) cases to test against, and if we run into bugs in the future we can always add a new case.

Reply:
rand is guaranteed by [posix](https://pubs.opengroup.org/onlinepubs/9699919799/functions/rand.html) to always return the same sequence when srand isn't used. I checked and our implementation is compliant.

> If srand() is then called with the same seed value, the sequence of pseudo-random numbers shall be repeated.
> If rand() is called before any calls to srand() are made, the same sequence shall be generated as when srand() is first called with a seed value of 1.
> The implementation shall behave as if no function defined in this volume of POSIX.1-2017 calls rand() or srand().

So the CI failures shouldn't be a concern. The auto generation of cases was invaluable when developing posix_memalign, so I would really prefer not to remove it, as it will be extremely useful for future malloc changes.

Reply:
I'll absolutely give you that randomizing test cases can be incredibly valuable while developing, but having these test cases depend on:
- the implementation of `srand()` and/or `rand()`;
- the order in which the tests are executed;
- the number of iterations of the individual tests;
- the implementation of the actual tests themselves;
- probably some other stuff I couldn't come up with :^)

...is probably a bit annoying for future development. Although without any changes in the test suite, CI should indeed not fail randomly.

Currently, I can't really review these test cases because it's not clear to me by reading the code what is actually tested in terms of memory alignment without executing the code and dumping the generated random values. Looking at the code you've written it should be pretty straightforward determining some edges cases to use as input, right?

All still my personal opinion so if you still decide against more explicit cases, I'll be glad to discuss this on #development or so!

Reply:
I agree with adding explicit test cases, and to also srand(1) at the start of the tests (I think currently each test is its own executable though). However, I would be uncomfortable removing this test, as I would be uncomfortable merging any malloc prs without running it.